src/RequestHeaderParser.php

```diff
@@ -158,7 +158,7 @@ private function parseRequest($headers)
             $parts = parse_url($request->getRequestTarget());
 
             // make sure value contains valid host component (IP or hostname), but no fragment
-            if (!isset($parts['scheme'], $parts['host']) || $parts['scheme'] !== 'http' || isset($parts['fragment'])) {
+            if ((!isset($parts['scheme'], $parts['host']) || $parts['scheme'] !== 'http' || isset($parts['fragment'])) && isset($parts['scheme'])) {
```

Comment:
This is now really hard to read and I'm not sure it does what it's supposed to do anymore. It looks like a `request-target` that starts with `://` will now pass this validation?

As an alternative, may I suggest reverting this and updating the above check instead? Your issue can only really happen for `origin-target` requests, which means we can actually skip this whole block if the `request-target` starts with a `/`.

Does this make sense to you?